ec/ecdatasource/deploymentsdatasource/schema.go

```diff
@@ -62,65 +79,133 @@ func (d *DataSource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnost
+		Computed:    true,
+		Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
+			"deployment_id": {
+				Type:        types.StringType,
+				Description: "The deployment unique ID.",
+				Computed:    true,
+			},
+			"name": {
+				Type:        types.StringType,
+				Description: "The name of the deployment.",
+				Computed:    true,
+			},
+			"alias": {
+				Type:        types.StringType,
+				Description: "Deployment alias.",
+				Computed:    true,
+			},
+			"elasticsearch_resource_id": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource unique ID.",
+				Computed:    true,
+			},
+			"elasticsearch_ref_id": {
+				Type:        types.StringType,
+				Description: "The Elasticsearch resource reference.",
+				Computed:    true,
+			},
+			"kibana_resource_id": {
+				Type:        types.StringType,
+				Description: "The Kibana resource unique ID.",
+				Computed:    true,
+			},
+			"kibana_ref_id": {
+				Type:        types.StringType,
+				Description: "The Kibana resource reference.",
+				Computed:    true,
+			},
+			"apm_resource_id": {
+				Type:        types.StringType,
+				Description: "The APM resource unique ID.",
+				Computed:    true,
+			},
+			"apm_ref_id": {
+				Type:        types.StringType,
+				Description: "The APM resource reference.",
+				Computed:    true,
+			},
+			"integrations_server_resource_id": {
+				Type:        types.StringType,
+				Description: "The Integrations Server resource unique ID.",
+				Computed:    true,
+			},
+			"integrations_server_ref_id": {
+				Type:        types.StringType,
+				Description: "The Integrations Server resource reference.",
+				Computed:    true,
+			},
+			"enterprise_search_resource_id": {
+				Type:        types.StringType,
+				Description: "The Enterprise Search resource unique ID.",
+				Computed:    true,
+			},
+			"enterprise_search_ref_id": {
+				Type:        types.StringType,
+				Description: "The Enterprise Search resource reference.",
+				Computed:    true,
+			},
+		}),
 	}
 }
 
 func deploymentAttrTypes() map[string]attr.Type {
-	return map[string]attr.Type{
-		"deployment_id":                   types.StringType,
-		"name":                            types.StringType,
-		"alias":                           types.StringType,
-		"elasticsearch_resource_id":       types.StringType,
-		"elasticsearch_ref_id":            types.StringType,
-		"kibana_resource_id":              types.StringType,
-		"kibana_ref_id":                   types.StringType,
-		"apm_resource_id":                 types.StringType,
-		"apm_ref_id":                      types.StringType,
-		"integrations_server_resource_id": types.StringType,
-		"integrations_server_ref_id":      types.StringType,
-		"enterprise_search_resource_id":   types.StringType,
-		"enterprise_search_ref_id":        types.StringType,
-	}
+	return deploymentsListSchema().Attributes.Type().(types.ListType).ElemType.(types.ObjectType).AttrTypes
```

Comment:
I understand why it's needed and don't know how it can be improved but it looks risky.
Also it maybe not so easy detectable if something goes wrong, e.g. if we change list type to set type. In that case I assume some test should crash. But can we easy detect the root cause in that case?
WDYT?

Reply:
The unit tests reliably break when this is wrong. 

```
panic: interface conversion: attr.Type is types.SetType, not types.ListType [recovered]
	panic: interface conversion: attr.Type is types.SetType, not types.ListType
…
github.com/elastic/terraform-provider-ec/ec/ecdatasource/deploymentsdatasource.deploymentAttrTypes()
	/Users/pascal/gec/terraform-provider-ec/ec/ecdatasource/deploymentsdatasource/schema.go:174 +0x110
…
```